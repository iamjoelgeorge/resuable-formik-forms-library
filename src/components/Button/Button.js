import React from 'react';

import styles from './Button.module.scss';

const Button = (props) => {
  const { label, type, onClick, ...rest } = props;

  return (
    <button className={styles.button} type={type} {...rest}>
      Submit
    </button>
  );
};

export default Button;
