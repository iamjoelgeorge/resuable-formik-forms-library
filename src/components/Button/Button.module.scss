@import '../../theme/variables.scss';

.button {
  border: none;
  padding: 10px;
  border-radius: 2px;
  background-color: $darkPurple;
  color: white;
  font-weight: bold;
  margin-top: 12px;

  &:disabled {
    background-color: rgb(89, 89, 89);
  }
}
