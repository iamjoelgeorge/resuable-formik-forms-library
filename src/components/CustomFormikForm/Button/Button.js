import React from 'react';

import PropTypes from 'prop-types';

import styles from './Button.module.scss';
import { joinClassNames } from '../../../utils/utils';
import {
  ExternalLinkIconWhite,
  ExternalLinkIconRed,
  ExternalLinkIconPurple,
} from '../../../constants/icons';

const Button = (props) => {
  const {
    variant,
    type,
    href,
    label,
    onClick,
    isDisabled,
    theme,
    showExternalLinkIcon,
    containerClass,
    formik,
    ...rest
  } = props;

  // const disableButton = type === 'submit' ? !(formik.dirty && formik.isValid) : isDisabled;
  const disableButton = type === 'submit' ? !formik.isValid : isDisabled;
  const goTo = !isDisabled ? href : '';

  const linkDisabledClass = isDisabled ? styles.linkDisabled : '';
  const linkAsButtonDisabledClass = isDisabled ? styles.linkButtonDisabled : '';

  const IconForLink = theme === 'purple' ? ExternalLinkIconPurple : ExternalLinkIconRed;

  const buttonClasses =
    theme === 'purple'
      ? joinClassNames([styles.button, styles.purpleTheme, containerClass])
      : joinClassNames([styles.button, styles.redTheme, containerClass]);

  const linkClasses =
    theme === 'purple'
      ? joinClassNames([styles.link, styles.purpleTheme, linkDisabledClass, containerClass])
      : joinClassNames([styles.link, styles.redTheme, linkDisabledClass, containerClass]);

  const linkAsButtonClasses = joinClassNames([
    buttonClasses,
    styles.linkAsButton,
    linkAsButtonDisabledClass,
  ]);

  const handleClickOnLink = (e) => {
    if (isDisabled) e.preventDefault();
  };

  const renderButton = () => {
    switch (variant) {
      case 'button':
        return (
          <button className={buttonClasses} type={type} disabled={disableButton} {...rest}>
            {label}
          </button>
        );

      case 'link':
        return (
          <a
            className={linkClasses}
            href={goTo}
            target='_blank'
            onClick={handleClickOnLink}
            rel='noreferrer'
          >
            <span>{label}</span>
            {showExternalLinkIcon && (
              <img className={styles.externalLinkIcon} src={IconForLink} alt='External Link' />
            )}
          </a>
        );

      case 'link_as_button':
        return (
          <a
            className={linkAsButtonClasses}
            href={goTo}
            target='_blank'
            onClick={handleClickOnLink}
            rel='noreferrer'
          >
            {label}
            {showExternalLinkIcon && (
              <img
                className={styles.externalLinkIcon}
                src={ExternalLinkIconWhite}
                alt='External Link'
              />
            )}
          </a>
        );

      default:
        return (
          <button
            className={buttonClasses}
            type={type}
            disabled={disableButton}
            onClick={onClick}
            {...rest}
          >
            {label}{' '}
            {showExternalLinkIcon && (
              <img
                className={styles.externalLinkIcon}
                src={ExternalLinkIconWhite}
                alt='External Link'
              />
            )}
          </button>
        );
    }
  };

  return <>{renderButton()}</>;
};

Button.propTypes = {
  variant: PropTypes.oneOf(['button', 'link', 'link_as_button']),
  type: PropTypes.oneOf(['submit', 'button']),
  href: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func,
  isDisabled: PropTypes.bool,
  theme: PropTypes.oneOf(['red', 'purple']),
  showExternalLinkIcon: PropTypes.bool,
  containerClass: PropTypes.string,
};

Button.defaultProps = {
  variant: 'button',
  type: 'submit',
  href: '#',
  label: '',
  onClick: () => {},
  isDisabled: false,
  theme: 'purple',
  showExternalLinkIcon: false,
  containerClass: '',
};

export default Button;
