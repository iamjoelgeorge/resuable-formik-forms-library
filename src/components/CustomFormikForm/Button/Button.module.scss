@import '../../../theme/variables.scss';
@import '../../../theme/utils.scss';

* {
  box-sizing: border-box;
}

@mixin useButtonTheme($bgColor, $hoverBgColor, $focusBgColor) {
  background-color: $bgColor;

  &:hover:not(:disabled),
  &:active {
    cursor: pointer;
    background-color: $hoverBgColor;
  }

  &:focus {
    border-color: $focusBgColor;
  }
}

@mixin useLinkTheme(
  $defaultTextColor,
  $hoverColor,
  $borderBottomColorOnFocus,
  $borderBottomColorOnActive,
  $textColorOnActive
) {
  color: $defaultTextColor;

  span {
    border: 2px solid transparent;
    padding-bottom: 2px;
  }

  &:hover:not(&.linkDisabled) {
    color: $hoverColor;
  }

  &:focus {
    span {
      border-bottom-style: dashed;
      border-bottom-color: $borderBottomColorOnFocus;
    }
  }

  &:active {
    border-color: $borderBottomColorOnActive;
    color: $textColorOnActive;
  }
}

.button,
.link {
  border: 3px solid transparent;
  border-radius: 4px;
  font-size: em(16);
  font-weight: $fontWeightExtraBold;
  line-height: 19px;
  margin-top: 12px;
  padding: 0 em(15);
  width: 198px;
  height: 51px;
  text-align: center;
}

.button {
  color: $white;
  transition: 0.1s all ease-in-out;
  outline: none;

  &.purpleTheme {
    @include useButtonTheme($primary, $darkPurple, $brightPurple);
  }

  &.redTheme {
    @include useButtonTheme($tooltipRed, $buttonHover, $pinkOne);
  }

  &:disabled {
    background-color: $white;
    color: $tertiaryGrey;
    border-color: $dGrey;
  }
}

.linkAsButton {
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;

  &.linkButtonDisabled {
    background-color: $white !important;
    cursor: not-allowed !important;
    color: $tertiaryGrey;
    border-color: $dGrey;
  }
}

.link {
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  outline: none;

  &.purpleTheme {
    @include useLinkTheme($primary, $darkPurple, $darkPurple, $brightPurple, $darkPurple);
  }

  &.redTheme {
    @include useLinkTheme($tooltipRed, $buttonHover, $tooltipRed, $pinkOne, $buttonHover);
  }

  &.linkDisabled {
    color: $tertiaryGrey;
    pointer-events: none;

    .externalLinkIcon {
      -webkit-filter: invert(50%); /* safari 6.0 - 9.0 */
      filter: invert(50%);
    }
  }
}

.externalLinkIcon {
  margin-left: em(8);
}
