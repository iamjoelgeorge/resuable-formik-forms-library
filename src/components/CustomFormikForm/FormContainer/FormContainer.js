import React, { Children, cloneElement } from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import { validateInput } from '../../../utils/validateInput';

const FormContainer = (props) => {
  const { initialValues, children, validations, containerClass, onSubmit } = props;

  const validationObject = {};
  validations.forEach((item) => {
    validateInput(item, validationObject);
  });

  const validationSchema = Yup.object().shape(validationObject);

  return (
    <Formik initialValues={initialValues} onSubmit={onSubmit} validationSchema={validationSchema}>
      {(formik) => {
        const childrenWithExtraProp = Children.map(children, (child) =>
          cloneElement(child, { formik }),
        );
        return <Form className={containerClass}>{childrenWithExtraProp}</Form>;
      }}
    </Formik>
  );
};

export default FormContainer;
