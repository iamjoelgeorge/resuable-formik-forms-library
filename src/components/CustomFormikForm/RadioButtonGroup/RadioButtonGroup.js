import React from 'react';

import { Field } from 'formik';
import PropTypes from 'prop-types';

import styles from './RadioButtonGroup.module.scss';
import ErrorText from '../ErrorText/ErrorText';
import SlidingLabel from '../SlidingLabel/SlidingLabel';
import ToolTip from '../ToolTip/ToolTip';
import AdditionalInfo from '../AdditionalInfo/AdditionalInfo';

const RadioButtonGroup = (props) => {
  const {
    name,
    options,
    mainLabel,
    formik,
    isRequired,
    isDisabled,
    helpLink,
    helpLinkText,
    tooltipLink,
    tooltipLinkText,
    optionalText,
    mainLabelTooltipBoxHeading,
    mainLabelTooltipBoxDescription,
    mainLabelTooltipBoxDescriptionElement,
    ...rest
  } = props;

  const { errors, values } = formik;

  const userHasVisitedTheInputField = formik.touched[name];
  const inputFieldHasErrors = errors[name];
  const addErrorClassesToLabelAndInput = !!userHasVisitedTheInputField && !!inputFieldHasErrors;

  const renderOptions = () =>
    options.map((option, index) => {
      const { tooltip } = option;

      const showOptionTooltipIcon =
        tooltip?.heading || tooltip?.description || tooltip?.customDescriptionElement;

      return (
        <div className={styles.radioButtonWithLabel} key={index}>
          <label className={styles.optionLabel}>
            <Field
              data-testid={`radio-button-${option.label}`}
              className={styles.radioInput}
              type='radio'
              name={name}
              value={option.value}
              disabled={isDisabled}
              checked={values[name]?.includes(option.value)}
              {...rest}
            />

            <span className={styles.presentationalRadioButton}>
              <span className={styles.radioButtonSelected} />
            </span>

            {option.label && <span className={styles.labelSpanElement}>{option.label}</span>}
          </label>

          {showOptionTooltipIcon && (
            <ToolTip
              heading={tooltip?.heading}
              description={tooltip?.description}
              descriptionElement={tooltip?.customDescriptionElement}
              containerClass={styles.tooltip}
            />
          )}
        </div>
      );
    });

  return (
    <div className={styles.container}>
      <SlidingLabel
        customClass={styles.componentHeading}
        label={mainLabel}
        inputEntered
        htmlFor={name}
        showErrorStyle={addErrorClassesToLabelAndInput}
        tooltipBoxHeading={mainLabelTooltipBoxHeading}
        tooltipBoxDescription={mainLabelTooltipBoxDescription}
        tooltipBoxDescriptionElement={mainLabelTooltipBoxDescriptionElement}
        inputIsRequired={isRequired}
      />

      <div className={styles.radioGroup} role='group' aria-labelledby={name}>
        {renderOptions()}
      </div>

      {addErrorClassesToLabelAndInput && (
        <ErrorText fieldName={name} containerClass={styles.error} />
      )}

      <AdditionalInfo
        optionalText={optionalText}
        helpLinkText={helpLinkText}
        helpLink={helpLink}
        tooltipLinkText={tooltipLinkText}
        tooltipLink={tooltipLink}
        customClass={styles.optionalText}
      />
    </div>
  );
};

RadioButtonGroup.propTypes = {
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      value: PropTypes.any,
      tooltip: PropTypes.shape({
        heading: PropTypes.string,
        description: PropTypes.string,
        customDescriptionElement: PropTypes.element,
      }),
    }),
  ),
  mainLabel: PropTypes.string,
  formik: PropTypes.shape({}),
  isRequired: PropTypes.bool,
  isDisabled: PropTypes.bool,
  helpLink: PropTypes.string,
  helpLinkText: PropTypes.string,
  tooltipLink: PropTypes.string,
  tooltipLinkText: PropTypes.string,
  optionalText: PropTypes.string,
  mainLabelTooltipBoxHeading: PropTypes.string,
  mainLabelTooltipBoxDescription: PropTypes.string,
  mainLabelTooltipBoxDescriptionElement: PropTypes.string,
};

RadioButtonGroup.defaultProps = {
  options: {},
  mainLabel: '',
  formik: {},
  isRequired: false,
  isDisabled: false,
  helpLink: '',
  helpLinkText: '',
  tooltipLink: '',
  tooltipLinkText: '',
  optionalText: '',
  mainLabelTooltipBoxHeading: '',
  mainLabelTooltipBoxDescription: '',
  mainLabelTooltipBoxDescriptionElement: null,
};

export default RadioButtonGroup;
