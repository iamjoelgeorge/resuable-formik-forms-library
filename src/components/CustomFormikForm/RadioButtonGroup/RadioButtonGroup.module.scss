@import '../../../theme/variables.scss';
@import '../../../theme/utils.scss';

.container {
  position: relative;
  margin-top: 30px;

  .radioInput {
    position: absolute;
    left: -999em;

    span.presentationalRadioButton {
      padding: 2px;
    }

    &:disabled + span.presentationalRadioButton,
    &:disabled + span.presentationalRadioButton + span.labelSpanElement {
      color: $secondaryGrey;
      cursor: not-allowed;
    }

    &:focus + span.presentationalRadioButton + span.labelSpanElement {
      border-color: $brightPurple;
    }
  }

  .radioGroup {
    display: flex;

    .radioButtonWithLabel {
      display: flex;

      &:not(:last-child) {
        margin-right: 20px;
      }
    }
  }

  .presentationalRadioButton {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 1px;
    width: em(16);
    height: em(16);
    margin-right: em(5);
    border: 1px solid $lighterDarkGrey;
    border-radius: 100%;
    box-shadow: 0 2px 2px $borderGrey;
  }

  .radioInput:checked + .presentationalRadioButton .radioButtonSelected {
    background-color: $brightPurple;
  }

  .focused {
    border-color: $brightPurple;
  }

  .radioButtonSelected {
    display: inline-block;
    width: 100%;
    height: 100%;
    border-radius: 100%;
  }

  .labelSpanElement {
    margin-top: em(3);
    padding-bottom: em(3);
    border: 0;
    border-bottom: 1px dashed transparent;
  }

  .componentHeading {
    top: -20px;
    left: 0;
  }

  .optionLabel {
    position: relative;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    font-size: em(14);
    line-height: 14px;
    width: fit-content;
    color: $primaryDark;
    cursor: pointer;
  }

  .tooltip {
    margin-left: 4px;

    img {
      width: 10px;
    }
  }

  .optionWithTooltip {
    display: flex;
  }

  .optionalText {
    margin-left: -10px;
  }
}
