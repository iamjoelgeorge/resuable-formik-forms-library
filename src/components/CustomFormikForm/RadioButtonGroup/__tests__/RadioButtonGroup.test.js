import React from 'react';

import { render, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import Button from '../../Button/Button';
import FormContainer from '../../FormContainer/FormContainer';
import RadioButtonGroup from '../../RadioButtonGroup/RadioButtonGroup';

const radioOptions = [
  { label: 'One Radio', value: 'one' },
  { label: 'Two Radio', value: 'two' },
  {
    label: 'Three Radio',
    value: 'three',
    tooltip: {
      heading: 'Heading',
      customDescriptionElement: (
        <p>
          I'm a custom description Element. Look me up on
          <a href='https://www.google.com'>Google</a>
        </p>
      ),
    },
  },
];

const defaultProps = {
  name: 'radioOption',
  options: radioOptions,
  mainLabel: 'radioOption',
  isRequired: false,
  isDisabled: false,
  helpLink: '',
  helpLinkText: '',
  tooltipLink: '',
  tooltipLinkText: '',
  optionalText: '',
  mainLabelTooltipBoxHeading: '',
  mainLabelTooltipBoxDescription: '',
  mainLabelTooltipBoxDescriptionElement: null,
};

const initialValues = {
  radioOption: '',
};

const defaultValidations = [];
const handleSubmit = jest.fn();

afterEach(cleanup);

describe('[Component]: RadioButtonGroup', () => {
  it('should match the Snapshot', () => {
    const tree = renderer
      .create(
        <FormContainer
          initialValues={initialValues}
          validations={defaultValidations}
          onSubmit={handleSubmit}
        >
          <RadioButtonGroup {...defaultProps} />
        </FormContainer>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render disabled checkboxes', () => {
    const props = {
      ...defaultProps,
      isDisabled: true,
    };

    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
      </FormContainer>,
    );

    radioOptions.forEach((option) => {
      const radioButton = getByTestId(`radio-button-${option.label}`);
      expect(radioButton).toHaveAttribute('disabled');
    });
  });

  it('should render the optional text', () => {
    const props = {
      ...defaultProps,
      optionalText: 'Test Optional text',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
      </FormContainer>,
    );

    expect(getByText('Test Optional text')).toBeTruthy();
  });

  it('should render the help link', () => {
    const props = {
      ...defaultProps,
      helpLinkText: 'Test help link',
      helpLink: 'https://virginaustralia.com/',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
      </FormContainer>,
    );

    expect(getByText('Test help link')).toBeTruthy();
  });

  it('should render the tooltip link', () => {
    const props = {
      ...defaultProps,
      tooltipLinkText: 'Test tooltip link',
      tooltipLink: 'https://virginaustralia.com/',
    };

    const { getByText } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
      </FormContainer>,
    );

    expect(getByText('Test tooltip link')).toBeTruthy();
  });

  it('should show the error component for a required radio button', async () => {
    const props = {
      ...defaultProps,
      isRequired: true,
    };

    const validations = [
      {
        name: 'radioOption',
        type: 'radio_button_group',
        isRequired: true,
        message: 'Please select at least one option',
      },
    ];

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={validations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
        <Button label='Submit' />
      </FormContainer>,
    );

    const submitButton = getByText('Submit');
    userEvent.click(submitButton);

    await waitFor(() => {
      expect(getByTestId('error-radioOption')).toHaveTextContent(
        'Please select at least one option',
      );
    });
  });

  it('should select a radio button by default', async () => {
    const initialValues = {
      radioOption: 'two',
    };

    const { getByText, getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...defaultProps} />
        <Button label='Submit' />
      </FormContainer>,
    );

    const checkedCheckbox = getByTestId('radio-button-Two Radio');
    expect(checkedCheckbox.checked).toEqual(true);
  });

  it('should render the Tooltip icon component for the main label', () => {
    const props = {
      ...defaultProps,
      mainLabelTooltipBoxHeading: 'Tooltip',
      mainLabelTooltipBoxDescription: 'Lorem Ipsum is simply dummy text.',
    };

    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...props} />
      </FormContainer>,
    );

    const toolTipComponent = getByTestId('tooltip-icon-Tooltip');
    expect(toolTipComponent).toBeTruthy();
  });

  it('should render the Tooltip icon component for the option label', () => {
    const { getByTestId } = render(
      <FormContainer
        initialValues={initialValues}
        validations={defaultValidations}
        onSubmit={handleSubmit}
      >
        <RadioButtonGroup {...defaultProps} />
      </FormContainer>,
    );

    radioOptions.forEach((item) => {
      if (item?.tooltip?.heading) {
        const tooltipComponent = getByTestId(`tooltip-icon-${item.tooltip.heading}`);
        expect(tooltipComponent).toBeTruthy();
      }
    });
  });
});
