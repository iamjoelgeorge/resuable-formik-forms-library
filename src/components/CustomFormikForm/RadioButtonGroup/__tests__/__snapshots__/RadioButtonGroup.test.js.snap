// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: RadioButtonGroup should match the Snapshot 1`] = `
<form
  action="#"
  className=""
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="container "
  >
    <div
      className="container   slideLabel componentHeading"
    >
      <label
        className="label"
        htmlFor="radioOption"
      >
        radioOption
      </label>
    </div>
    <div
      aria-labelledby="radioOption"
      className="radioGroup"
      role="group"
    >
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={false}
            className="radioInput"
            data-testid="radio-button-One Radio"
            disabled={false}
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "getFieldHelpers": [Function],
                "getFieldMeta": [Function],
                "getFieldProps": [Function],
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialErrors": Object {},
                "initialStatus": undefined,
                "initialTouched": Object {},
                "initialValues": Object {
                  "radioOption": "",
                },
                "isSubmitting": false,
                "isValid": true,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validateOnMount": false,
                "values": Object {
                  "radioOption": "",
                },
              }
            }
            name="radioOption"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="one"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            One Radio
          </span>
        </label>
      </div>
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={false}
            className="radioInput"
            data-testid="radio-button-Two Radio"
            disabled={false}
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "getFieldHelpers": [Function],
                "getFieldMeta": [Function],
                "getFieldProps": [Function],
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialErrors": Object {},
                "initialStatus": undefined,
                "initialTouched": Object {},
                "initialValues": Object {
                  "radioOption": "",
                },
                "isSubmitting": false,
                "isValid": true,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validateOnMount": false,
                "values": Object {
                  "radioOption": "",
                },
              }
            }
            name="radioOption"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="two"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            Two Radio
          </span>
        </label>
      </div>
      <div
        className="radioButtonWithLabel"
      >
        <label
          className="optionLabel"
        >
          <input
            checked={false}
            className="radioInput"
            data-testid="radio-button-Three Radio"
            disabled={false}
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "getFieldHelpers": [Function],
                "getFieldMeta": [Function],
                "getFieldProps": [Function],
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialErrors": Object {},
                "initialStatus": undefined,
                "initialTouched": Object {},
                "initialValues": Object {
                  "radioOption": "",
                },
                "isSubmitting": false,
                "isValid": true,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validateOnMount": false,
                "values": Object {
                  "radioOption": "",
                },
              }
            }
            name="radioOption"
            onBlur={[Function]}
            onChange={[Function]}
            type="radio"
            value="three"
          />
          <span
            className="presentationalRadioButton"
          >
            <span
              className="radioButtonSelected"
            />
          </span>
          <span
            className="labelSpanElement"
          >
            Three Radio
          </span>
        </label>
        <div
          className="container tooltip"
          data-testid="tooltip-icon-Heading"
        >
          <span
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <img
              alt="tooltip"
              src="test-file-stub"
            />
          </span>
        </div>
      </div>
    </div>
    
    
    
  </div>
</form>
`;
